sigmastate/src/test/scala/sigmastate/utxo/DistributedSigSpecification.scala

```diff
@@ -481,4 +486,32 @@ class DistributedSigSpecification extends SigmaTestingCommons {
     verifier.verify(prop, ctx, validProofB, fakeMessage).get._1 shouldBe true
   }
 
+  property("distributed message signing - AND (2 out of 2)") {
+    val proverA = new ErgoLikeTestProvingInterpreter
+    val proverB = new ErgoLikeTestProvingInterpreter
+    val verifier: ContextEnrichingTestProvingInterpreter = new ContextEnrichingTestProvingInterpreter
+
+    val msg = "Let's have a deal".getBytes("UTF-8")
+
+    val pubkeyAlice = proverA.dlogSecrets.head.publicImage
+    val pubkeyBob = proverB.dlogSecrets.head.publicImage
+
+    val sigmaTree = CAND(Seq(pubkeyAlice, pubkeyBob))
+
+    val hintsFromBob: HintsBag = proverB.generateCommitments(sigmaTree)
+    val bagA = HintsBag(hintsFromBob.realCommitments)
+
+    val sigAlice = proverA.signMessage(sigmaTree, msg, bagA).get
+
+    val bagB = proverB.bagForMultisig(ctx, sigmaTree, sigAlice, Seq(pubkeyAlice))
+      .addHint(hintsFromBob.ownCommitments.head)
+
+    val sigBob = proverB.signMessage(sigmaTree, msg, bagB).get
+
+    // Proof generated by Alice without getting Bob's part is not correct
+    verifier.verifySignature(sigmaTree, msg, sigAlice) shouldBe false
+
+    // Compound proof from Bob is correct
+    verifier.verifySignature(sigmaTree, msg, sigBob) shouldBe true
+  }
```

Comment:
Looks like all the scenarios can be generalized to test both signing, and old-style prove/verify. It seams the data flow is almost the same. 
So I suggest to add verifySignature calls into all the existing cases.

Reply:
verifySignature is already inside verify(), for transaction bytes as a message.

To bring another (custom) message into all the tests, we need to separate data and execution via an interpreter pattern (e.g. Free monad), but this will reduce readability , and an advantage is unclear.

Reply:
Oh, no Free Monad please.